apex-squid/src/test/java/org/fundacionjala/enforce/sonarqube/apex/parser/grammar/ApexGrammarQueryExpressionTest.java

```diff
@@ -0,0 +1,482 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2016 Fundacion Jala.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.fundacionjala.enforce.sonarqube.apex.parser.grammar;
+
+import org.junit.Test;
+import org.junit.Before;
+
+import org.fundacionjala.enforce.sonarqube.apex.parser.ApexRuleTest;
+
+import static org.fundacionjala.enforce.sonarqube.apex.api.grammar.ApexGrammarRuleKey.QUERY;
+import static org.sonar.sslr.tests.Assertions.assertThat;
+
+public class ApexGrammarQueryExpressionTest extends ApexRuleTest {
+
+    @Before
+    public void init() {
+        setRootRule(QUERY);
+    }
+
+    /**
+     * SELECT SENTENCE TESTS
+     * 
+     */
+    @Test
+    public void positiveRules_SubQueryExpression() {
+        assertThat(parser)
+                .matches("SELECT field1,field2, "
+                        + "(SELECT sub1 FROM tab2 LIMIT 200) "
+                        + "FROM table1 LIMIT 2000")
+                .matches("SELECT field1,field2, (SELECT sub1 FROM tab2 LIMIT 2) "
+                        + "FROM table1 LIMIT 2000");
+    }
+    
+    @Test
+    public void negativeRules_SubQueryExpression() {
+        assertThat(parser)
+                .notMatches("SELECT field1,field2, "
+                        + "(SELECT sub1 FROM LIMIT 200) "
+                        + "FROM table1 LIMIT 2000")
+                .notMatches("SELECT field1,field2, (SELECT FROM tab2 LIMIT 2) "
+                        + "FROM table1 LIMIT 2000");
+    }
+    
+     @Test
+    public void positiveRules_CountQueryExpression() {
+        assertThat(parser)
+                .matches("SELECT COUNT(dato) field1,field2, "
```

Comment:
this is not valid
.matches("SELECT COUNT(dato) field1,field2, "

.matches("SELECT COUNT(dato), field1, field2, "